Dart/src/com/jetbrains/lang/dart/ide/formatter/DartSpacingProcessor.java

```diff
@@ -371,14 +349,41 @@ else if (type1 == COLON) {
     }
 
     if (elementType == VALUE_EXPRESSION && type2 == CASCADE_REFERENCE_EXPRESSION) {
-      //if (type1 == CASCADE_REFERENCE_EXPRESSION) {
-      //  if (((AbstractBlock)child1).getNode().getLastChildNode().getElementType() == CALL_EXPRESSION) {
-      //    // TODO if child1 message name == child2 message name, do not add line break
-      //  }
-      //}
+      if (type1 == CASCADE_REFERENCE_EXPRESSION) {
+        if (cascadesAreSameMethod(((AbstractBlock)child1).getNode(), ((AbstractBlock)child2).getNode())) {
+          return Spacing.createSpacing(0, 1, 0, mySettings.KEEP_LINE_BREAKS, mySettings.KEEP_BLANK_LINES_IN_CODE);
```

Comment:
The 2nd argument (maxSpaces) should probably be zero. Dart Style doesn't allow spaces before cascade when it is on the same line.
The same about line 361 below.